pennylane/qchem/structure.py

```diff
@@ -412,3 +413,136 @@ def excitations_to_wires(singles, doubles, wires=None):
         doubles_wires.append([d1_wires, d2_wires])
 
     return singles_wires, doubles_wires
+
+
+def pubchem_data(identifier, identifier_type="name"):
```

Comment:
How come the name of this function is `pubchem_data` but the docstring calls it as `mol_data_from_pubchem` ?

Reply:
I like the name `mol_data()`

In fact, I wonder if we could expande `molecular_hamiltonian` such that it could also take just the descriptor of the molecule, e.g. `BeH2`, and then return the Hamiltonian

Reply:
😍

Reply:
Minor point but I think the name pubchem might be better to appear in the function name.

Reply:
I don't like having pubchem in the function name. I prefer to have it in the first line of the doc instead